File: .NET -네트워크/C# Window Form/1018_채팅클라이언트/1018_채팅클라이언트/통신/Packet.cs
//packet.cs

namespace _1018_채팅클라이언트
{
    static class Packet
    {
        const string PACK_SHORTMESSAGE      = "shortmessage";

        public static string ShortMessage(string nickname, string msg)
        {
            string packet = string.Empty;

            packet += PACK_SHORTMESSAGE + "\a";
            packet += nickname + "#";
            packet += msg;

            return packet;
        }
       
    }
}

File: WIN-32 System/5_Client/packet.cpp
#include "std.h"

//서로 주고받는 데이터(packet) 정의


PACKET packet_CreatePacket(HWND hWnd, const TCHAR* nickName, const TCHAR* msg)
{
	PACKET pk;

	pk.hWnd = hWnd;
	_tcscpy_s(pk.nickName, _countof(pk.nickName), nickName);
	_tcscpy_s(pk.msg, _countof(pk.msg), msg);
	GetLocalTime(&pk.dt);

	return pk;
}

PACKET packet_CreatePacket(HWND hWnd, const TCHAR* nickName, const TCHAR* msg, SYSTEMTIME dt)
{
	PACKET pk;

	pk.hWnd = hWnd;
	_tcscpy_s(pk.nickName, _countof(pk.nickName), nickName);
	_tcscpy_s(pk.msg, _countof(pk.msg), msg);
	pk.dt = dt;

	return pk;
}

